File: GeneratedEntityReaderClasses/Components/ElectricChargeComponent.cs
namespace GeneratedNoitaClasses;
public class ElectricChargeComponent : NoitaComponentBase,  iNoitaType<ElectricChargeComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt charge_time_frames { get; set; } = new();
	public NoitaFloat fx_velocity_max { get; set; } = new();
	public NoitaInt electricity_emission_interval_frames { get; set; } = new();
	public NoitaInt fx_emission_interval_min { get; set; } = new();
	public NoitaInt fx_emission_interval_max { get; set; } = new();
	public NoitaInt charge { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		charge_time_frames.Read(s);
		fx_velocity_max.Read(s);
		electricity_emission_interval_frames.Read(s);
		fx_emission_interval_min.Read(s);
		fx_emission_interval_max.Read(s);
		charge.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		charge_time_frames.Write(s);
		fx_velocity_max.Write(s);
		electricity_emission_interval_frames.Write(s);
		fx_emission_interval_min.Write(s);
		fx_emission_interval_max.Write(s);
		charge.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ElectricChargeComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class ElectricChargeComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    charge_time_frames:NoitaInt = field(default_factory=NoitaInt)
    fx_velocity_max:NoitaFloat = field(default_factory=NoitaFloat)
    electricity_emission_interval_frames:NoitaInt = field(default_factory=NoitaInt)
    fx_emission_interval_min:NoitaInt = field(default_factory=NoitaInt)
    fx_emission_interval_max:NoitaInt = field(default_factory=NoitaInt)
    charge:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.charge_time_frames.Read(s)
        self.fx_velocity_max.Read(s)
        self.electricity_emission_interval_frames.Read(s)
        self.fx_emission_interval_min.Read(s)
        self.fx_emission_interval_max.Read(s)
        self.charge.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.charge_time_frames.Write(s)
        self.fx_velocity_max.Write(s)
        self.electricity_emission_interval_frames.Write(s)
        self.fx_emission_interval_min.Write(s)
        self.fx_emission_interval_max.Write(s)
        self.charge.Write(s)

